### PR TITLE
curvefs/tool: list-topology output json file

### LINKED ISSUE #975: curvefs/tool: list-topology output jsonfile
**Is your feature request related to a problem? (你需要的功能是否与某个问题有关?)**

list-topology output jsonfile wich can be used to create-topology

**Describe the solution you'd like (描述你期望的解决方法)**

use the output of  list-topology to build a json file

**Describe alternatives you've considered (描述你想到的折衷方案)**

**Additional context/screenshots (更多上下文/截图)**

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
add output json file which can be used for create-topology

### What problem does this PR solve?

Issue Number: close #975  

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Cyber-SiKu** [comment]
recheck

File: curvefs/src/tools/curvefs_tool_define.h [hunk 1]
```diff
@@ -133,6 +133,14 @@ const char kHostFollowerValue[] = "follower";
 const char kEtcdLeaderValue[] = "StateLeader";
 const char kEtcdFollowerValue[] = "StateFollower";
 
+/* fs type */
+const char kFsTypeS3[] = "s3";
+const char kFsTypeVolume[] = "volume";
+
+/* json type */
+const char kJsonTypeBuild[] = "build";
```
Comment (SeanHai) [inline curvefs/src/tools/curvefs_tool_define.h:141]: What's the difference of json type of 'build' and 'tree'?

Comment (Cyber-SiKu) [reply curvefs/src/tools/curvefs_tool_define.h:141]: > What's the difference of json type of 'build' and 'tree'?

build jsonfile can be used to build topology,
tree json file will form a tree of topology according to the structure of cluser, pool, zone, server, metaserver

**Review by SeanHai — APPROVED** [review]

File: curvefs/src/tools/list/curvefs_topology_list.h [hunk 2]
```diff
@@ -133,36 +137,64 @@ class TopologyListTool
...
+
+    bool GetClusterTree(Json::Value* cluster, const std::string& clusterId);
+
+    bool GetPoolTree(Json::Value* pool, uint64_t poolId);
+
+    bool GetZoneTree(Json::Value* zone, uint64_t zoneId);
+
+    bool GetServerTree(Json::Value* server, uint64_t serverId);
+
+    bool GetMetaserverTree(Json::Value* metaserver, uint64_t metaserverId);
```
Comment (wu-hanqing) [inline curvefs/src/tools/list/curvefs_topology_list.h:154]: keep a class small, so this format logic should not be put in here.
condier define tree builder / json builder into single class

File: curvefs/src/tools/list/curvefs_topology_list.h [hunk 3]
```diff
@@ -133,36 +137,64 @@ class TopologyListTool
...
     std::map<mds::topology::MetaServerIdType, MetaserverInfoType>
-        metaserverId2MetaserverInfo;
+        metaserverId2MetaserverInfo_;
+
+    /**
+     * @brief cluster map, like conf/topology.json
+     *
+     * @details
+     */
+    Json::StreamWriterBuilder clusterMap_;
```
[the remark this answers is not in the document]
Comment (Cyber-SiKu) [reply curvefs/src/tools/list/curvefs_topology_list.h:197]: > there's no need to define it as a data member

It is reserved to add a return function if other commands need to use it in the future, but it is also a good idea to modify it at that time.

File: curvefs/src/tools/list/curvefs_topology_list.h (hunk 2 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/list/curvefs_topology_list.h:154]: > keep a class small, so this format logic should not be put in here. condier define tree builder / json builder into single class

fix

**Review by wu-hanqing — APPROVED** [review]

**Pull request merged**